service_provider.go

```diff
@@ -378,6 +378,40 @@ func (ivr *InvalidResponseError) Error() string {
 	return fmt.Sprintf("Authentication failed")
 }
 
+func responseIsSigned(response *etree.Document) (bool, error) {
```

Comment:
There was a comment here in #113 about whether this is correct when the response is encrypted (https://github.com/crewjam/saml/pull/113#discussion_r140833636).

As I read the SAML Core spec, individual parts of the response can be encrypted, but not the whole message. So if both a response signature and encryption are present, the signature should still sit outside any encrypted elements. In any case, it looks like this library only supports encrypted assertions.